src/cli.rs

```diff
@@ -142,6 +144,10 @@ pub enum Commands {
         #[arg(short, value_parser = valid_kmer, default_value_t = DEFAULT_KMER)]
         k: usize,
 
+        /// Number of reads before stopping
+        #[arg(long)]
+        proportion_reads: Option<usize>,
```

Comment:
I think this should be an f64, not usize. And you should add a parser which checks it's between 0 and 1.